sympy/stats/crv_types.py

```diff
@@ -1260,6 +1261,13 @@ def _cdf(self, x):
         return Piecewise((uppergamma(a,b/x)/gamma(a), x > 0),
                         (S.Zero, True))
 
+    def sample(self):
+        try :
+            from scipy.stats import invgamma
```

Comment:
SymPy has `import_module` in sympy.external.importtools, which checks for ImportError and many other things like incompatible versions. Use it. This try-except block should be replaced by 
```
scipy = import_module('scipy')
if scipy:
    from scipy.stats import invgamma
```
(import_module returns None if it failed.)
See https://github.com/sympy/sympy/blob/master/sympy/external/importtools.py